doc: add suggestions regarding Cargo flags for Release builds

zamazan4ik opened the pull request:
Partially resolves https://github.com/ratatui/ratatui-website/issues/930

In this PR, I want to introduce additional guidelines for Ratatui users regarding recommended Cargo options for their Ratatui-based apps. The recommendations are highly inspired by the similar Tauri [documentation](https://v2.tauri.app/concept/size/#cargo-configuration) (no licensing issues since Tauri docs are MIT licensed).

My motivation for that:

* By enabling better compiler optimizations, we can deliver better user experience for end users of Ratatui-based applications
* Enabling LTO, CG1 (`codegen-units = 1`) and `strip` reduces the resulting binary size - pretty common complaint about Rust apps. So we can improve the situation a bit on the ecosystem level

My personal motivation: I use a lot of Rust (and Ratatui) based apps in various environments, and some of my environments are pretty size-constrained. My main use-case for smaller apps is preparing minimal Docker images (especially when we talk about minimal sidecars).

Differences from Tauri recommendations:

* I removed potentially harmful recommendations like `panic = abort`. I am not saying that this is a bad recommendation - just saying this setting is a bit more dangerous than enabling LTO or `codegen-units`. If we are ok with that - no problem, we can also add a line about this setting too
* I left recommendations just for the stable Rustc. If we are interested enough in giving recommendations for Nightly-only stuff - we can add it later.

Maybe someone can find arguable a recommendation about `strip` setting. In this case, we can remove it from the default recommendation. However, personally, I think we should recommend it by default too.

I am open for the discussion about the recommended defaults for Ratatui.

I didn't test how my changes are rendered locally - just created this PR via GitHub web UI.

Commit:
doc: add suggestions regarding Cargo flags for Release builds

- add recommended Cargo options for Ratatui-based based for their Release builds

orhun commented on src/content/docs/recipes/apps/release-your-app.md:
```diff
@@ -31,6 +31,30 @@ Discord server / forum. This is not a comprehensive list, but it should help you
 [Ratatui Discord]: https://discord.gg/pMCEU9hNEj
 [Ratatui Forum]: https://forum.ratatui.rs
 
+## Cargo configuration
```
I would love to move this to the end of the documentation, maybe under "Optimizations" section and reference it from the Release checklist above. The rationale is to keep the checklist & screenshots under primary focus and provide additional improvements later on. What do you think?

zamazan4ik replied:
I like the idea! Just pushed such a variant ;)

orhun reviewed (commented):
Thanks for the PR!

Commit:
doc: move optimizations to the page end

- resolves comments from https://github.com/ratatui/ratatui-website/pull/931/files#r2314325552

orhun reviewed (approved):
Thanks!
